src/backend/executor/nodeHash.c

```diff
@@ -770,8 +770,17 @@ ExecHashIncreaseNumBatches(HashJoinTable hashtable)
 	{
 		Size		sz = nbatch * sizeof(stats->batchstats[0]);
 
+		/*
+		 * We use repalloc_huge because the condition in the beginning
+		 * assumes that oldnbatches and nbatches will be used to
+		 * index values of size 8 or less (size of void*), but the
+		 * size of HashJoinBatchStats structure is 80 bytes, so even
+		 * if we pass that check, this still might not fit in the
+		 * MaxAllocSize. The maximum amount of memory we can request
+		 * here is ~10GB.
+		 */
 		stats->batchstats =
-			(HashJoinBatchStats *) repalloc(stats->batchstats, sz);
+			(HashJoinBatchStats *) repalloc_huge(stats->batchstats, sz);
```

Comment:
How about additional [check for 32-bit machine](https://github.com/arenadata/gpdb/commit/263865a48973767ce8ed7b7788059a38a24a9f37#diff-5b735c65568e935c0bdf768116c93c1266ac7b4c516a06fb1aa847b4e093cff1R1049-R1051)?

Reply:
Done